<!doctype html>
<html class="docs-version-current" lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.13">
<link rel="search" type="application/opensearchdescription+xml" title="Nirvana" href="/NirvanaDocumentation/opensearch.xml">
<link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Open+Sans&family=Raleway&family=Source+Code+Pro&display=swap"><title data-react-helmet="true">Canonical Transcripts | Nirvana</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:url" content="https://illumina.github.io/NirvanaDocumentation/core-functionality/canonical-transcripts"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="current"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Canonical Transcripts | Nirvana"><meta data-react-helmet="true" name="description" content="Overview"><meta data-react-helmet="true" property="og:description" content="Overview"><link data-react-helmet="true" rel="icon" href="/NirvanaDocumentation/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://illumina.github.io/NirvanaDocumentation/core-functionality/canonical-transcripts"><link data-react-helmet="true" rel="alternate" href="https://illumina.github.io/NirvanaDocumentation/core-functionality/canonical-transcripts" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://illumina.github.io/NirvanaDocumentation/core-functionality/canonical-transcripts" hreflang="x-default"><link data-react-helmet="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link rel="stylesheet" href="/NirvanaDocumentation/assets/css/styles.d26686ac.css">
<link rel="preload" href="/NirvanaDocumentation/assets/js/runtime~main.4bbb35e8.js" as="script">
<link rel="preload" href="/NirvanaDocumentation/assets/js/main.8a147b1e.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#" class="skipToContent_OuoZ">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/NirvanaDocumentation/"><div class="navbar__logo"><img src="/NirvanaDocumentation/img/NirvanaLogo.svg" alt="Nirvana" class="themedImage_TMUO themedImage--light_4Vu1"><img src="/NirvanaDocumentation/img/NirvanaLogo.svg" alt="Nirvana" class="themedImage_TMUO themedImage--dark_uzRr"></div></a></div><div class="navbar__items navbar__items--right"><div class="navbar__item dropdown dropdown--hoverable dropdown--right"><a class="navbar__link" href="/NirvanaDocumentation/">3.18 (unreleased)</a><ul class="dropdown__menu"><li><a aria-current="page" class="dropdown__link dropdown__link--active" href="/NirvanaDocumentation/core-functionality/canonical-transcripts">3.18 (unreleased)</a></li><li><a class="dropdown__link" href="/NirvanaDocumentation/3.17/core-functionality/canonical-transcripts">3.17</a></li><li><a class="dropdown__link" href="/NirvanaDocumentation/3.16/core-functionality/canonical-transcripts">3.16</a></li><li><a class="dropdown__link" href="/NirvanaDocumentation/3.14/">3.14</a></li><li><a class="dropdown__link" href="/NirvanaDocumentation/versions">All versions</a></li></ul></div><a href="https://github.com/Illumina/Nirvana" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><div class="searchBox_Utm0"><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span></span><span class="DocSearch-Button-Keys"></span></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docs-wrapper docs-doc-page"><div class="docPage_lDyR"><button aria-label="Scroll back to top" class="clean-btn theme-back-to-top-button backToTopButton_i9tI" type="button"></button><aside class="docSidebarContainer_0YBq"><div class="sidebar_a3j0"><nav class="menu thin-scrollbar menu_cyFh"><ul class="theme-doc-sidebar-menu menu__list"><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist hasHref_TwRn" href="/NirvanaDocumentation/">Introduction</a></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist hasHref_TwRn" href="/NirvanaDocumentation/data-sources/1000Genomes">Data Sources</a></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist hasHref_TwRn" href="/NirvanaDocumentation/file-formats/nirvana-json-file-format">File Formats</a></div></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist menu__link--active hasHref_TwRn" href="/NirvanaDocumentation/core-functionality/canonical-transcripts">Core Functionality</a></div><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/NirvanaDocumentation/core-functionality/canonical-transcripts">Canonical Transcripts</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/NirvanaDocumentation/core-functionality/gene-fusions">Gene Fusion Detection</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/NirvanaDocumentation/core-functionality/mnv-recomposition">MNV Recomposition</a></li><li class="theme-doc-sidebar-item-link theme-doc-sidebar-item-link-level-2 menu__list-item"><a class="menu__link" tabindex="0" href="/NirvanaDocumentation/core-functionality/variant-ids">Variant IDs</a></li></ul></li><li class="theme-doc-sidebar-item-category theme-doc-sidebar-item-category-level-1 menu__list-item menu__list-item--collapsed"><div class="menu__list-item-collapsible"><a class="menu__link menu__link--sublist hasHref_TwRn" href="/NirvanaDocumentation/utilities/jasix">Utilities</a></div></li></ul></nav></div></aside><main class="docMainContainer_r8cw"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_zHA2"><div class="docItemContainer_oiyr"><article><span class="theme-doc-version-badge badge badge--secondary">Version: <!-- -->3.18 (unreleased)</span><div class="tocCollapsible_aw-L theme-doc-toc-mobile tocMobile_Tx6Y"><button type="button" class="clean-btn tocCollapsibleButton_zr6a">On this page</button></div><div class="theme-doc-markdown markdown"><header><h1>Canonical Transcripts</h1></header><h2 class="anchor anchorWithStickyNavbar_y2LR" id="overview">Overview<a aria-hidden="true" class="hash-link" href="#overview" title="Direct link to heading">​</a></h2><p>One of the more polarizing topics within annotation is the notion of canonical transcripts. Because of alternative splicing, we often have several transcripts for each gene. In the human genome, there are an average of 3.4 transcripts per gene (Tung, 2020). As scientists, we seem to have a need for identifying a representative example of a gene - even if there&#x27;s no biological basis for the motivation.</p><p><img src="/NirvanaDocumentation/assets/images/hk1-transcripts-a5b85474d3b002553687715dbd004907.png"></p><div class="admonition admonition-info alert alert--info"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="14" height="16" viewBox="0 0 14 16"><path fill-rule="evenodd" d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg></span>Golden Helix Blog</h5></div><div class="admonition-content"><p>A few years ago, the guys over at Golden Helix wrote an excellent post about the pitfalls and issues surrounding the identification of canonical transcripts: <a href="https://blog.goldenhelix.com/whats-in-a-name-the-intricacies-of-identifying-variants/" target="_blank" rel="noopener noreferrer">What’s in a Name: The Intricacies of Identifying Variants</a>.</p></div></div><p>In Nirvana, we wanted to identify an algorithm for determining the canonical transcript and apply it consistently to all of our transcript data sources.</p><h2 class="anchor anchorWithStickyNavbar_y2LR" id="known-algorithms">Known Algorithms<a aria-hidden="true" class="hash-link" href="#known-algorithms" title="Direct link to heading">​</a></h2><h3 class="anchor anchorWithStickyNavbar_y2LR" id="ucsc">UCSC<a aria-hidden="true" class="hash-link" href="#ucsc" title="Direct link to heading">​</a></h3><p>UCSC publishes a list of canonical transcripts in its <code>knownCanonical</code> table which is available via the <a href="https://genome.ucsc.edu/cgi-bin/hgTables" target="_blank" rel="noopener noreferrer">TableBrowser</a>. Of the RefSeq data sources, it was the only one we could find that provided canonical transcripts:</p><blockquote><p>The canonical transcript is defined as either the longest CDS, if the gene has translated transcripts, or the longest cDNA.</p></blockquote><p>If you were to implement this and compare it with the knownCanonical table, you would see a lot of exceptions to the rule.</p><h3 class="anchor anchorWithStickyNavbar_y2LR" id="ensembl">Ensembl<a aria-hidden="true" class="hash-link" href="#ensembl" title="Direct link to heading">​</a></h3><p>The <a href="http://uswest.ensembl.org/Help/Glossary" target="_blank" rel="noopener noreferrer">Ensembl glossary</a> states:</p><blockquote><p>The canonical transcript is used in the gene tree analysis in Ensembl and does not necessarily reflect the most biologically relevant transcript of a gene. For human, the canonical transcript for a gene is set according to the following hierarchy:</p><ol><li>Longest CCDS translation with no stop codons.</li><li>If no (1), choose the longest Ensembl/Havana merged translation with no stop codons.</li><li>If no (2), choose the longest translation with no stop codons.</li><li>If no translation, choose the longest non-protein-coding transcript.</li></ol></blockquote><h3 class="anchor anchorWithStickyNavbar_y2LR" id="acmg">ACMG<a aria-hidden="true" class="hash-link" href="#acmg" title="Direct link to heading">​</a></h3><p>From the ACMG Guidelines for the Interpretation of Sequence Variants:</p><blockquote><p>A reference transcript for each gene should be used and provided in the report when describing coding variants. The transcript should represent either the longest known transcript and/or the most clinically relevant transcript.</p></blockquote><h3 class="anchor anchorWithStickyNavbar_y2LR" id="clinvar">ClinVar<a aria-hidden="true" class="hash-link" href="#clinvar" title="Direct link to heading">​</a></h3><p>From the ClinVar paper:</p><blockquote><p>When there are multiple transcripts for a gene, ClinVar selects one HGVS expression to construct a preferred name. By default, this selection is based on the first reference standard transcript identified by the RefSeqGene/LRG (Locus Reference Genomic) collaboration.</p></blockquote><h2 class="anchor anchorWithStickyNavbar_y2LR" id="unified-approach">Unified Approach<a aria-hidden="true" class="hash-link" href="#unified-approach" title="Direct link to heading">​</a></h2><p>Our approach is almost identical to the one Golden Helix discussed in their article:</p><ol><li>If we&#x27;re looking at RefSeq, only consider NM &amp; NR transcripts as candidates for canonical transcripts.</li><li>Sort the transcripts in the following order:<ol><li><a href="https://www.lrg-sequence.org/" target="_blank" rel="noopener noreferrer">Locus Reference Genomic (LRG)</a> entries occur before non-LRG entries</li><li>Descending CDS length</li><li>Descending transcript length</li><li>Ascending accession number</li></ol></li><li>Grab the first entry</li></ol></div><footer class="theme-doc-footer docusaurus-mt-lg"><div class="theme-doc-footer-edit-meta-row row"><div class="col"><a href="https://github.com/Illumina/NirvanaDocumentation/edit/master/docs/core-functionality/canonical-transcripts.md" target="_blank" rel="noreferrer noopener" class="theme-edit-this-page"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_mS5F" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_mt2f"></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/NirvanaDocumentation/file-formats/custom-annotations"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">« <!-- -->Custom Annotations</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/NirvanaDocumentation/core-functionality/gene-fusions"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Gene Fusion Detection<!-- --> »</div></a></div></nav></div></div><div class="col col--3"><div class="tableOfContents_vrFS thin-scrollbar theme-doc-toc-desktop"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#overview" class="table-of-contents__link toc-highlight">Overview</a></li><li><a href="#known-algorithms" class="table-of-contents__link toc-highlight">Known Algorithms</a><ul><li><a href="#ucsc" class="table-of-contents__link toc-highlight">UCSC</a></li><li><a href="#ensembl" class="table-of-contents__link toc-highlight">Ensembl</a></li><li><a href="#acmg" class="table-of-contents__link toc-highlight">ACMG</a></li><li><a href="#clinvar" class="table-of-contents__link toc-highlight">ClinVar</a></li></ul></li><li><a href="#unified-approach" class="table-of-contents__link toc-highlight">Unified Approach</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="footer__bottom text--center"><div class="footer__copyright">© 2022 Illumina, Inc. All rights reserved.</div></div></div></footer></div>
<script src="/NirvanaDocumentation/assets/js/runtime~main.4bbb35e8.js"></script>
<script src="/NirvanaDocumentation/assets/js/main.8a147b1e.js"></script>
</body>
</html>