<!doctype html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-alpha.66">
<link rel="preconnect" href="https://www.google-analytics.com">
<link rel="preconnect" href="https://www.googletagmanager.com">
<script async src="https://www.googletagmanager.com/gtag/js?id=G-5KXNW9LCD7"></script>
<script>function gtag(){dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-5KXNW9LCD7",{})</script>
<link rel="search" type="application/opensearchdescription+xml" title="Nirvana" href="/NirvanaDocumentation/opensearch.xml">
<link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Open+Sans&family=Raleway&family=Source+Code+Pro&display=swap"><title data-react-helmet="true">Custom Annotations | Nirvana</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="current"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Custom Annotations | Nirvana"><meta data-react-helmet="true" name="description" content="Overview"><meta data-react-helmet="true" property="og:description" content="Overview"><meta data-react-helmet="true" property="og:url" content="https://illumina.github.io/NirvanaDocumentation/file-formats/custom-annotations"><link data-react-helmet="true" rel="shortcut icon" href="/NirvanaDocumentation/img/favicon.ico"><link data-react-helmet="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link data-react-helmet="true" rel="canonical" href="https://illumina.github.io/NirvanaDocumentation/file-formats/custom-annotations"><link rel="stylesheet" href="/NirvanaDocumentation/styles.88ee2314.css">
<link rel="preload" href="/NirvanaDocumentation/styles.5391dede.js" as="script">
<link rel="preload" href="/NirvanaDocumentation/runtime~main.ffb1032a.js" as="script">
<link rel="preload" href="/NirvanaDocumentation/main.0901f0a9.js" as="script">
<link rel="preload" href="/NirvanaDocumentation/1.1f2ef902.js" as="script">
<link rel="preload" href="/NirvanaDocumentation/2.2104805c.js" as="script">
<link rel="preload" href="/NirvanaDocumentation/24.61398d78.js" as="script">
<link rel="preload" href="/NirvanaDocumentation/27.d643c295.js" as="script">
<link rel="preload" href="/NirvanaDocumentation/935f2afb.f984beab.js" as="script">
<link rel="preload" href="/NirvanaDocumentation/17896441.a0b73652.js" as="script">
<link rel="preload" href="/NirvanaDocumentation/1c2d50a1.4947fa1b.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><div aria-label="Navigation bar toggle" class="navbar__toggle" role="button" tabindex="0"><svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 30 30" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></div><a class="navbar__brand" href="/NirvanaDocumentation/"><img class="navbar__logo" src="/NirvanaDocumentation/img/NirvanaLogo.svg"></a></div><div class="navbar__items navbar__items--right"><div class="navbar__item dropdown dropdown--hoverable dropdown--right"><a class="navbar__item navbar__link" href="/NirvanaDocumentation/">3.13 (unreleased)</a><ul class="dropdown__menu"><li><a aria-current="page" class="dropdown__link dropdown__link--active" href="/NirvanaDocumentation/file-formats/custom-annotations">3.13 (unreleased)</a></li><li><a class="dropdown__link" href="/NirvanaDocumentation/3.12/">3.12</a></li><li><a class="dropdown__link" href="/NirvanaDocumentation/versions">All versions</a></li></ul></div><a href="https://github.com/Illumina/Nirvana" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span><span class="DocSearch-Button-Key">⌘</span><span class="DocSearch-Button-Key">K</span></button></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div><div class="navbar-sidebar"><div class="navbar-sidebar__brand"><a class="navbar__brand" href="/NirvanaDocumentation/"><img class="navbar__logo" src="/NirvanaDocumentation/img/NirvanaLogo.svg"></a></div><div class="navbar-sidebar__items"><div class="menu"><ul class="menu__list"><li class="menu__list-item"><a role="button" class="menu__link menu__link--sublist">Versions</a><ul class="menu__list"><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active" href="/NirvanaDocumentation/file-formats/custom-annotations">3.13 (unreleased)</a></li><li class="menu__list-item"><a class="menu__link" href="/NirvanaDocumentation/3.12/">3.12</a></li><li class="menu__list-item"><a class="menu__link" href="/NirvanaDocumentation/versions">All versions</a></li></ul></li><li class="menu__list-item"><a href="https://github.com/Illumina/Nirvana" target="_blank" rel="noopener noreferrer" class="menu__link header-github-link" aria-label="GitHub repository"></a></li></ul></div></div></div></nav><div class="main-wrapper"><div class="docPage_2gpo"><div class="docSidebarContainer_3_JD" role="complementary"><div class="sidebar_2urC"><div class="menu menu--responsive menu_5FrY"><button aria-label="Open Menu" aria-haspopup="true" class="button button--secondary button--sm menu__button" type="button"><svg aria-label="Menu" class="sidebarMenuIcon_Dm3K" xmlns="http://www.w3.org/2000/svg" height="24" width="24" viewBox="0 0 32 32" role="img" focusable="false"><title>Menu</title><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><ul class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Introduction</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/NirvanaDocumentation/">Introduction</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/NirvanaDocumentation/introduction/dependencies">Dependencies</a></li><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/NirvanaDocumentation/introduction/getting-started">Getting Started</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#!">Data Sources</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="-1" href="/NirvanaDocumentation/data-sources/clinvar">ClinVar</a></li></ul></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#!">File Formats</a><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/NirvanaDocumentation/file-formats/nirvana-json-file-format">Nirvana JSON File Format</a></li><li class="menu__list-item"><a aria-current="page" class="menu__link menu__link--active active" tabindex="0" href="/NirvanaDocumentation/file-formats/custom-annotations">Custom Annotations</a></li></ul></li></ul></div></div></div><main class="docMainContainer_3EyW"><div class="container padding-vert--lg docItemWrapper_1EkI"><div class="row"><div class="col docItemCol_2ASc"><div class="docItemContainer_3QWW"><article><div><span class="badge badge--secondary">Version: 3.13 (unreleased)</span></div><header><h1 class="docTitle_1Lrw">Custom Annotations</h1></header><div class="markdown"><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="overview"></a>Overview<a aria-hidden="true" tabindex="-1" class="hash-link" href="#overview" title="Direct link to heading">#</a></h2><p>While the team tries to keep data sources up-to-date, you might want to start incorporate new annotations ahead of our update cycle. Another
common use case involves protected health information (PHI). Custom annotations are a mechanism that enables both use cases.</p><p>Here are some examples of how our collaborators use custom annotations:</p><ul><li>associating context from both a patient-level and a patient cohort level with the variant annotations</li><li>adding content that is licensed (e.g. HGMD) to the variant annotations</li></ul><p>At the moment, we have two different custom annotation file formats. One provides additional annotations to variants (both small variants and SVs)
while the other caters to gene annotations.</p><p>In both cases, the custom annotation file format is a tab-delimited file that is separated into two parts: the header &amp; the data.</p><p>The header is where you can customize how you want the data to appear in the JSON file and provide context about the genome assembly and how
Nirvana should match the variants.</p><p>At Illumina, there are usually many components downstream of Nirvana that have to parse our annotations. If a customer provides a custom
annotation, those downstream tools need to understand more about the data such as:</p><ul><li>data type (e.g. number, boolean, or a string)</li><li>data category (e.g. is this an allele count, allele number, allele frequency, etc.)</li><li>associated population (i.e. if this is an allele frequency)</li></ul><p>For each custom annotation, Nirvana uses this context to create a <a href="https://json-schema.org/" target="_blank" rel="noopener noreferrer">JSON schema</a> that can be sent to downstream tools. If
a tool knows that this is an allele frequency, it can validate user input to ensure that it&#x27;s in the range of [0, 1].</p><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="variant-file-format"></a>Variant File Format<a aria-hidden="true" tabindex="-1" class="hash-link" href="#variant-file-format" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="basic-allele-frequency-example"></a>Basic Allele Frequency Example<a aria-hidden="true" tabindex="-1" class="hash-link" href="#basic-allele-frequency-example" title="Direct link to heading">#</a></h3><p>Imagine that you want to create a basic allele frequency custom annotation for small variants. It would look something like this:</p><table><thead><tr><th align="left">Column 1</th><th align="left">2</th><th align="left">3</th><th align="left">4</th><th align="left">5</th></tr></thead><tbody><tr><td align="left">#title=MyDataSource</td><td align="left"></td><td align="left"></td><td align="left"></td><td align="left"></td></tr><tr><td align="left">#assembly=GRCh37</td><td align="left"></td><td align="left"></td><td align="left"></td><td align="left"></td></tr><tr><td align="left">#matchVariantsBy=allele</td><td align="left"></td><td align="left"></td><td align="left"></td><td align="left"></td></tr><tr><td align="left">#CHROM</td><td align="left">POS</td><td align="left">REF</td><td align="left">ALT</td><td align="left">allAf</td></tr><tr><td align="left">#categories</td><td align="left">.</td><td align="left">.</td><td align="left">.</td><td align="left">AlleleFrequency</td></tr><tr><td align="left">#descriptions</td><td align="left">.</td><td align="left">.</td><td align="left">.</td><td align="left">ALL</td></tr><tr><td align="left">#type</td><td align="left">.</td><td align="left">.</td><td align="left">.</td><td align="left">number</td></tr><tr><td align="left">chr1</td><td align="left">100</td><td align="left">A</td><td align="left">C</td><td align="left">0.000159</td></tr><tr><td align="left">chr1</td><td align="left">102</td><td align="left">T</td><td align="left">G</td><td align="left">0.000325</td></tr><tr><td align="left">chr1</td><td align="left">106</td><td align="left">A</td><td align="left">T</td><td align="left">0.001421</td></tr></tbody></table><p>Let&#x27;s go over the header and discuss the contents:</p><ul><li><code>title</code> indicates the name of the JSON key</li><li><code>assembly</code> indicates that this data is only valid for <code>GRCh37</code></li><li><code>matchVariantsBy</code> indicates that we should only match the annotations if they are allele-specific</li><li><code>categories</code> provides hints to downstream tools on how they might want to treat the data. In this case, we indicate that it&#x27;s an allele
frequency.</li><li><code>descriptions</code> are used in special circumstances to provide more context. Even though column 5 is called <code>allAf</code>, it might not be clear to a
downstream tool that this means a global allele frequency using all sub-populations. In this case, <code>ALL</code> indicates the intended population.</li><li><code>type</code> indicates to downstream tools the data type. Since allele frequencies are numbers, we&#x27;ll write <code>number</code> in this column.</li></ul><p>If we converted this custom annotation to Nirvana format and ran Nirvana on the following VCF file:</p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><button tabindex="0" type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button><div class="prism-code language-undefined codeBlock_3iAC"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">##fileformat=VCFv4.1</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">#CHROM POS ID REF ALT QUAL FILTER INFO</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">chr1    100 .   A   C   .   .   .</span></div></div></div></div></div><p>We would expect the following data to show up in our JSON output file:</p><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><button tabindex="0" type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button><div class="prism-code language-json codeBlock_3iAC"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">   </span><span class="token property">&quot;MyDataSource&quot;</span><span class="token operator" style="color:rgb(137, 221, 255)">:</span><span class="token punctuation" style="color:rgb(199, 146, 234)">[</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">      </span><span class="token punctuation" style="color:rgb(199, 146, 234)">{</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">         </span><span class="token property">&quot;refAllele&quot;</span><span class="token operator" style="color:rgb(137, 221, 255)">:</span><span class="token string" style="color:rgb(195, 232, 141)">&quot;A&quot;</span><span class="token punctuation" style="color:rgb(199, 146, 234)">,</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">         </span><span class="token property">&quot;altAllele&quot;</span><span class="token operator" style="color:rgb(137, 221, 255)">:</span><span class="token string" style="color:rgb(195, 232, 141)">&quot;C&quot;</span><span class="token punctuation" style="color:rgb(199, 146, 234)">,</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">         </span><span class="token property">&quot;allAf&quot;</span><span class="token operator" style="color:rgb(137, 221, 255)">:</span><span class="token number" style="color:rgb(247, 140, 108)">0.000159</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">      </span><span class="token punctuation" style="color:rgb(199, 146, 234)">}</span><span class="token plain"></span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">   </span><span class="token punctuation" style="color:rgb(199, 146, 234)">]</span></div></div></div></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="gene-file-format"></a>Gene File Format<a aria-hidden="true" tabindex="-1" class="hash-link" href="#gene-file-format" title="Direct link to heading">#</a></h2><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="customizing-the-header"></a>Customizing the Header<a aria-hidden="true" tabindex="-1" class="hash-link" href="#customizing-the-header" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="title"></a>Title<a aria-hidden="true" tabindex="-1" class="hash-link" href="#title" title="Direct link to heading">#</a></h3><p>For the title, you can provide any string that hasn&#x27;t already been used. The title should be unique.</p><div class="admonition admonition-caution alert alert--warning"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</h5></div><div class="admonition-content"><p>Make sure that the title does not conflict with other keys in the JSON file.</p></div></div><p>For small variants, you can&#x27;t provide a title that conflicts with other keys in the variant object. Some examples of this would be
<code>vid</code>, <code>chromosome</code>, <code>transcripts</code>, etc.. The title should also not conflict with other data source keys like <code>clinvar</code> or <code>gnomad</code>.</p><p>For structural variants, you can&#x27;t provide a title that conflicts with other keys in the position object. Some examples of this would be
<code>chromosome</code>, <code>svLength</code>, <code>cytogeneticBand</code>, etc. The title should also not conflict with other data source keys like <code>clingen</code> or <code>dgv</code>.</p><div class="admonition admonition-caution alert alert--warning"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"></path></svg></span>caution</h5></div><div class="admonition-content"><p>Care should be taken not to annotate using multiple custom annotations that all use the same title.</p></div></div><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="genome-assemblies"></a>Genome Assemblies<a aria-hidden="true" tabindex="-1" class="hash-link" href="#genome-assemblies" title="Direct link to heading">#</a></h3><p>The following genome assemblies can be specified:</p><ul><li>GRCh37</li><li>GRCh38</li></ul><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="matching-criteria"></a>Matching Criteria<a aria-hidden="true" tabindex="-1" class="hash-link" href="#matching-criteria" title="Direct link to heading">#</a></h3><p>The matching criteria instructs how Nirvana should match a VCF variant to the custom annotation.</p><p>The following matching criteria can be specified:</p><ul><li><code>allele</code> - use this when you only want allele-specific matches. This is commonly the case when using allele frequency data sources like <code>gnomAD</code></li><li><code>position</code> - use this when you want positional matches. This is commonly used with disease phenotype data sources like <code>ClinVar</code></li><li><code>sv</code> - use this when you want to match to all other overlapping SVs. This use case arose when we were adding custom annotations for baseline
copy number intervals along the genome.</li></ul><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="categories"></a>Categories<a aria-hidden="true" tabindex="-1" class="hash-link" href="#categories" title="Direct link to heading">#</a></h3><p>Categories are not used by Nirvana, but are often used by downstream tools. Categories provide hints for how those tools should filter or display
the annotation data.</p><p>When a category is specified, Nirvana will provide additional validation for those fields. The following table describes each category:</p><table><thead><tr><th align="left">Category</th><th align="left">Description</th><th align="left">Validation</th></tr></thead><tbody><tr><td align="left">AlleleCount</td><td align="left">allele counts for a specific population</td><td align="left">See the supported populations below</td></tr><tr><td align="left">AlleleNumber</td><td align="left">allele numbers for a specific population</td><td align="left">See the supported populations below</td></tr><tr><td align="left">AlleleFrequency</td><td align="left">allele frequencies for a specific population</td><td align="left">See the supported populations below</td></tr><tr><td align="left">Prediction</td><td align="left">ACMG-style pathogenicity classifications</td><td align="left">• <code>benign</code> (B)<br>• <code>likely benign</code> (LB)<br>• <code>VUS</code><br>• <code>likely pathogenic</code> (LP)<br>• <code>pathogenic</code> (P)</td></tr><tr><td align="left">Filter</td><td align="left">free text that signals downstream tools to add the column to the filter</td><td align="left">Max 20 characters</td></tr><tr><td align="left">Description</td><td align="left">free-text description</td><td align="left">Max 100 characters</td></tr><tr><td align="left">Identifier</td><td align="left">any ID</td><td align="left">Max 50 characters</td></tr><tr><td align="left">HomozygousCount</td><td align="left">count of homozygous individuals for a specific population</td><td align="left">See the supported populations below</td></tr><tr><td align="left">Score</td><td align="left">any score value</td><td align="left">Any double-precision floating point number</td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="descriptions"></a>Descriptions<a aria-hidden="true" tabindex="-1" class="hash-link" href="#descriptions" title="Direct link to heading">#</a></h3><p>Descriptions are used to add more context to the categories. For now, descriptions are mainly used to associate allele counts, numbers, and frequencies with their respective populations.</p><h4><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="populations"></a>Populations<a aria-hidden="true" tabindex="-1" class="hash-link" href="#populations" title="Direct link to heading">#</a></h4><p>The following populations were specified in the HapMap project, 1000 Genomes Project, ExAC, and gnomAD.</p><table><thead><tr><th align="left">Population Code</th><th align="left">Super-population Code</th><th align="left">Description</th></tr></thead><tbody><tr><td align="left">ACB</td><td align="left">AFR</td><td align="left">African Caribbeans in Barbados</td></tr><tr><td align="left">AFR</td><td align="left">AFR</td><td align="left">African</td></tr><tr><td align="left">ALL</td><td align="left">ALL</td><td align="left">All populations</td></tr><tr><td align="left">AMR</td><td align="left">AMR</td><td align="left">Ad Mixed American</td></tr><tr><td align="left">ASJ</td><td align="left"></td><td align="left">Ashkenazi Jewish</td></tr><tr><td align="left">ASW</td><td align="left">AFR</td><td align="left">Americans of African Ancestry in SW USA</td></tr><tr><td align="left">BEB</td><td align="left">SAS</td><td align="left">Bengali from Bangladesh</td></tr><tr><td align="left">CDX</td><td align="left">EAS</td><td align="left">Chinese Dai in Xishuangbanna, China</td></tr><tr><td align="left">CEU</td><td align="left">EUR</td><td align="left">Utah Residents (CEPH) with Northern and Western European Ancestry</td></tr><tr><td align="left">CHB</td><td align="left">EAS</td><td align="left">Han Chinese in Beijing, China</td></tr><tr><td align="left">CHS</td><td align="left">EAS</td><td align="left">Southern Han Chinese</td></tr><tr><td align="left">CLM</td><td align="left">AMR</td><td align="left">Colombians from Medellin, Colombia</td></tr><tr><td align="left">EAS</td><td align="left">EAS</td><td align="left">East Asian</td></tr><tr><td align="left">ESN</td><td align="left">AFR</td><td align="left">Esan in Nigeria</td></tr><tr><td align="left">EUR</td><td align="left">EUR</td><td align="left">European</td></tr><tr><td align="left">FIN</td><td align="left">EUR</td><td align="left">Finnish in Finland</td></tr><tr><td align="left">GBR</td><td align="left">EUR</td><td align="left">British in England and Scotland</td></tr><tr><td align="left">GIH</td><td align="left">SAS</td><td align="left">Gujarati Indian from Houston, Texas</td></tr><tr><td align="left">GWD</td><td align="left">AFR</td><td align="left">Gambian in Western Divisions in the Gambia</td></tr><tr><td align="left">IBS</td><td align="left">EUR</td><td align="left">Iberian population in Spain</td></tr><tr><td align="left">ITU</td><td align="left">SAS</td><td align="left">Indian Telugu from the UK</td></tr><tr><td align="left">JPT</td><td align="left">EAS</td><td align="left">Japanese in Tokyo, Japan</td></tr><tr><td align="left">KHV</td><td align="left">EAS</td><td align="left">Kinh in Ho Chi Minh City, Vietnam</td></tr><tr><td align="left">LWK</td><td align="left">AFR</td><td align="left">Luhya in Webuye, Kenya</td></tr><tr><td align="left">MAG</td><td align="left">AFR</td><td align="left">Mandinka in the Gambia</td></tr><tr><td align="left">MKK</td><td align="left">AFR</td><td align="left">Maasai in Kinyawa, Kenya</td></tr><tr><td align="left">MSL</td><td align="left">AFR</td><td align="left">Mende in Sierra Leone</td></tr><tr><td align="left">MXL</td><td align="left">AMR</td><td align="left">Mexican Ancestry from Los Angeles, USA</td></tr><tr><td align="left">NFE</td><td align="left">EUR</td><td align="left">European (Non-Finnish)</td></tr><tr><td align="left">OTH</td><td align="left">OTH</td><td align="left">Other</td></tr><tr><td align="left">PEL</td><td align="left">AMR</td><td align="left">Peruvians from Lima, Peru</td></tr><tr><td align="left">PJL</td><td align="left">SAS</td><td align="left">Punjabi from Lahore, Pakistan</td></tr><tr><td align="left">PUR</td><td align="left">AMR</td><td align="left">Puerto Ricans from Puerto Rico</td></tr><tr><td align="left">SAS</td><td align="left">SAS</td><td align="left">South Asian</td></tr><tr><td align="left">STU</td><td align="left">SAS</td><td align="left">Sri Lankan Tamil from the UK</td></tr><tr><td align="left">TSI</td><td align="left">EUR</td><td align="left">Toscani in Italia</td></tr><tr><td align="left">YRI</td><td align="left">AFR</td><td align="left">Yoruba in Ibadan, Nigeria</td></tr></tbody></table><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="data-types"></a>Data Types<a aria-hidden="true" tabindex="-1" class="hash-link" href="#data-types" title="Direct link to heading">#</a></h3><p>Each custom annotation can be one of the following data types:</p><ul><li><code>bool</code> - true or false</li><li><code>number</code> - any integer or floating-point number</li><li><code>string</code> - text</li></ul><div class="admonition admonition-tip alert alert--success"><div class="admonition-heading"><h5><span class="admonition-icon"><svg xmlns="http://www.w3.org/2000/svg" width="12" height="16" viewBox="0 0 12 16"><path fill-rule="evenodd" d="M6.5 0C3.48 0 1 2.19 1 5c0 .92.55 2.25 1 3 1.34 2.25 1.78 2.78 2 4v1h5v-1c.22-1.22.66-1.75 2-4 .45-.75 1-2.08 1-3 0-2.81-2.48-5-5.5-5zm3.64 7.48c-.25.44-.47.8-.67 1.11-.86 1.41-1.25 2.06-1.45 3.23-.02.05-.02.11-.02.17H5c0-.06 0-.13-.02-.17-.2-1.17-.59-1.83-1.45-3.23-.2-.31-.42-.67-.67-1.11C2.44 6.78 2 5.65 2 5c0-2.2 2.02-4 4.5-4 1.22 0 2.36.42 3.22 1.19C10.55 2.94 11 3.94 11 5c0 .66-.44 1.78-.86 2.48zM4 14h5c-.23 1.14-1.3 2-2.5 2s-2.27-.86-2.5-2z"></path></svg></span>tip</h5></div><div class="admonition-content"><p>For boolean variables, only keys with a <code>true</code> value will be output to the JSON object.</p></div></div><h2><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="using-sautils"></a>Using SAUtils<a aria-hidden="true" tabindex="-1" class="hash-link" href="#using-sautils" title="Direct link to heading">#</a></h2><p>Nirvana includes a tool called <code>SAUtils</code> that converts various data sources into Nirvana&#x27;s native binary format. The sub-commands <code>customvar</code> and <code>customgene</code> are used to specify a variant file or a gene file respectively.</p><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="convert-variant-file"></a>Convert Variant File<a aria-hidden="true" tabindex="-1" class="hash-link" href="#convert-variant-file" title="Direct link to heading">#</a></h3><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><button tabindex="0" type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button><div class="prism-code language-Bash codeBlock_3iAC"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">dotnet bin/Release/netcoreapp2.1/SAUtils.dll customvar \</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">     -r Data/References/Homo_sapiens.GRCh37.Nirvana.dat \</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">     -i MyDataSource.tsv \</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">     -o SupplementaryAnnotation</span></div></div></div></div></div><ul><li>the <code>-r</code> argument specifies the compressed reference path</li><li>the <code>-i</code> argument specifies the input TSV path</li><li>the <code>-o</code> argument specifies the output directory</li></ul><h3><a aria-hidden="true" tabindex="-1" class="anchor enhancedAnchor_2cZh" id="convert-gene-file"></a>Convert Gene File<a aria-hidden="true" tabindex="-1" class="hash-link" href="#convert-gene-file" title="Direct link to heading">#</a></h3><div class="mdxCodeBlock_1XEh"><div class="codeBlockContent_1u-d"><button tabindex="0" type="button" aria-label="Copy code to clipboard" class="copyButton_10dd">Copy</button><div class="prism-code language-Bash codeBlock_3iAC"><div class="codeBlockLines_b7E3" style="color:#bfc7d5;background-color:#292d3e"><div class="token-line" style="color:#bfc7d5"><span class="token plain">dotnet bin/Release/netcoreapp2.1/SAUtils.dll customgene \</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">     --uga Nirvana_UGA.tsv \</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">     -i MyDataSource.tsv \</span></div><div class="token-line" style="color:#bfc7d5"><span class="token plain">     -o SupplementaryAnnotation</span></div></div></div></div></div><ul><li>the <code>--uga</code> argument specifies the Nirvana universal gene archive (UGA) path</li><li>the <code>-i</code> argument specifies the input TSV path</li><li>the <code>-o</code> argument specifies the output directory</li></ul></div></article><div class="margin-vert--xl"><div class="row"><div class="col"><a href="https://github.com/Illumina/NirvanaDocumentation/edit/master/docs/file-formats/custom-annotations.mdx" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="1.2em" width="1.2em" preserveAspectRatio="xMidYMid meet" viewBox="0 0 40 40" style="margin-right:0.3em;vertical-align:sub"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div></div></div><div class="margin-vert--lg"><nav class="pagination-nav" aria-label="Blog list page navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/NirvanaDocumentation/file-formats/nirvana-json-file-format"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">« Nirvana JSON File Format</div></a></div><div class="pagination-nav__item pagination-nav__item--next"></div></nav></div></div></div><div class="col col--3"><div class="tableOfContents_3SO_"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#overview" class="table-of-contents__link">Overview</a></li><li><a href="#variant-file-format" class="table-of-contents__link">Variant File Format</a><ul><li><a href="#basic-allele-frequency-example" class="table-of-contents__link">Basic Allele Frequency Example</a></li></ul></li><li><a href="#gene-file-format" class="table-of-contents__link">Gene File Format</a></li><li><a href="#customizing-the-header" class="table-of-contents__link">Customizing the Header</a><ul><li><a href="#title" class="table-of-contents__link">Title</a></li><li><a href="#genome-assemblies" class="table-of-contents__link">Genome Assemblies</a></li><li><a href="#matching-criteria" class="table-of-contents__link">Matching Criteria</a></li><li><a href="#categories" class="table-of-contents__link">Categories</a></li><li><a href="#descriptions" class="table-of-contents__link">Descriptions</a></li><li><a href="#data-types" class="table-of-contents__link">Data Types</a></li></ul></li><li><a href="#using-sautils" class="table-of-contents__link">Using SAUtils</a><ul><li><a href="#convert-variant-file" class="table-of-contents__link">Convert Variant File</a></li><li><a href="#convert-gene-file" class="table-of-contents__link">Convert Gene File</a></li></ul></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="text--center"><div>© 2020 Illumina, Inc. All rights reserved.</div></div></div></footer></div>
<script src="/NirvanaDocumentation/styles.5391dede.js"></script>
<script src="/NirvanaDocumentation/runtime~main.ffb1032a.js"></script>
<script src="/NirvanaDocumentation/main.0901f0a9.js"></script>
<script src="/NirvanaDocumentation/1.1f2ef902.js"></script>
<script src="/NirvanaDocumentation/2.2104805c.js"></script>
<script src="/NirvanaDocumentation/24.61398d78.js"></script>
<script src="/NirvanaDocumentation/27.d643c295.js"></script>
<script src="/NirvanaDocumentation/935f2afb.f984beab.js"></script>
<script src="/NirvanaDocumentation/17896441.a0b73652.js"></script>
<script src="/NirvanaDocumentation/1c2d50a1.4947fa1b.js"></script>
</body>
</html>